app/quest-boost/[boostId]/page.tsx

```diff
@@ -0,0 +1,192 @@
+    if (!signature) return;
+    setSign(signature);
+  };
+
+  useEffect(() => {
+    if (!account || !boost || sign[0].length === 0 || sign[1].length === 0)
+      return;
+
+    const callContract = async () => {
+      const amount = uint256.bnToUint256(boost.amount);
+      const claimCallData = CallData.compile({
+        amount: amount,
+        token: boost.token,
+        boost_id: boost.id,
+        signature: sign,
+      });
+
+      const { transaction_hash } = await account.execute(
+        {
+          contractAddress: process.env.NEXT_PUBLIC_QUEST_BOOST_CONTRACT ?? "",
+          entrypoint: "claim",
+          calldata: claimCallData,
+        },
+        undefined,
+        { maxFee: 900_000_000_000_000 }
+      );
+
+      if (transaction_hash) {
+        addTransaction({
+          timestamp: Date.now(),
+          subtext: boost?.name ?? "Quest Boost Rewards",
+          type: NotificationType.TRANSACTION,
+          data: {
+            type: TransactionType.CLAIM_REWARDS,
+            hash: transaction_hash,
+            status: "pending",
+          },
+        });
+      }
+    };
+
+    callContract();
+  }, [sign]);
+
+  return (
+    <div className={styles.container}>
+      <div className="flex flex-col">
+        <h1 className={styles.title}>{boost?.name}</h1>
+        {boost?.expiry ? (
+          <Timer fixed={false} expiry={Number(boost?.expiry)} />
+        ) : null}
+      </div>
+
+      <div className={styles.card_container}>
+        {quests?.map((quest, index) => {
+          if (quest?.hidden || quest?.disabled) return null;
+          return (
+            <Quest
+              key={index}
+              title={quest.title_card}
+              onClick={() => router.push(`/quest/${quest.id}`)}
+              imgSrc={quest.img_card}
+              issuer={{
+                name: quest.issuer,
+                logoFavicon: quest.logo,
+              }}
+              reward={quest.rewards_title}
+              id={quest.id}
+              expired={quest.expired}
+            />
+          );
+        })}
+      </div>
+      <div className={styles.claim_button_container}>
+        <div className={styles.claim_button_text_content}>
+          <p>Reward:</p>
+          <div className="flex flex-row gap-2">
+            <p className={styles.claim_button_text_highlight}>
+              {boost?.amount} USDC
+            </p>
+            <CDNImage
+              src={"/icons/usdc.svg"}
+              priority
+              width={32}
+              height={32}
+              alt="usdc icon"
+            />
+          </div>
+          <p>among</p>
+          <p className={styles.claim_button_text_highlight}>
+            {participants} players
+          </p>
+        </div>
+        <div>
+          <Button
+            disabled={boost?.winner !== hexToDecimal(address)}
+            onClick={handleClaimClick}
+          >
+            {boost?.winner === hexToDecimal(address)
```

Comment:
I would prefer something like that, chained ? operators are not easy to read:
```tsx
        {(() => {
            if (boost?.winner === hexToDecimal(address)) {
                return "Claim boost reward 🎉 ";
            } else if (boost?.expiry > Date.now()) {
                return "Claim boost reward 🎉 ";
            } else {
                return "You’re not selected 🙁";
            }
        })()}
        ```
        
        Also the message should probably not be Claim boost reward if the expiry is higher, but just that the boost has not ended

Reply:
done